actions/contacts/upsert.go

```diff
@@ -0,0 +1,61 @@
+package contacts
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/bitcoin-sv/spv-wallet/engine"
+	"github.com/bitcoin-sv/spv-wallet/mappings"
+	"github.com/bitcoin-sv/spv-wallet/models"
+	"github.com/bitcoin-sv/spv-wallet/server/auth"
+	"github.com/gin-gonic/gin"
+)
+
+// upsert will add a new contact or modify an existing one.
+// Upsert contact godoc
+// @Summary		Upsert contact
+// @Description	Add or update contact. For new contact send request to add current user as contact
+// @Tags		Contact
+// @Produce		json
+// @Param		paymail path string true "Paymail address of the contact the user wants to add/modify"
+// @Param		UpsertContact body contacts.UpsertContact true "Full name and metadata needed to add/modify contact"
+// @Success		201
+// @Router		/v1/contact/:paymail [PUT]
```

Comment:
Here it should be still:
`// @Router		/v1/contact/{paymail} [PUT]`

Reply:
fixed